common/config/data.ts

```diff
@@ -256,43 +256,29 @@ export const NODES: { [key: string]: NodeConfig } = {
   }
 };
 
-export function initWeb3Node(): Promise<void> {
-  return new Promise((resolve, reject) => {
-    const { web3 } = window as any;
+export async function initWeb3Node(): Promise<void> {
+  const { web3 } = window as any;
 
-    if (!web3) {
-      return reject(
-        new Error(
-          'Web3 not found. Please check that MetaMask is installed, or that MyEtherWallet is open in Mist.'
-        )
-      );
-    }
+  if (!web3 || !web3.currentProvider || !web3.currentProvider.sendAsync) {
+    throw new Error(
+      'Web3 not found. Please check that MetaMask is installed, or that MyEtherWallet is open in Mist.'
+    );
+  }
 
-    if (web3.version.network === 'loading') {
-      return reject(
-        new Error(
-          'MetaMask / Mist is still loading. Please refresh the page and try again.'
-        )
-      );
-    }
+  const lib = new Web3Node();
+  const networkId = await lib.getNetworkId();
 
-    web3.version.getNetwork((err, networkId) => {
-      if (err) {
-        return reject(err);
-      }
+  if (networkId === 'loading') {
+    throw new Error(
+      'MetaMask / Mist is still loading. Please refresh the page and try again.'
+    );
+  }
 
-      try {
-        NODES.web3 = {
-          network: networkIdToName(networkId),
-          service: 'MetaMask / Mist',
-          lib: new Web3Node(web3),
-          estimateGas: false,
-          hidden: true
-        };
-        resolve();
-      } catch (err) {
-        reject(err);
-      }
-    });
-  });
+  NODES.web3 = {
+    network: networkIdToName(networkId),
+    service: 'MetaMask / Mist',
+    lib,
+    estimateGas: false,
+    hidden: true
+  };
```

Comment:
I didn't realize that this was how this worked, mutating `NODES`. Is there any problem with components being out of date by the time this happens?

Reply:
I'm not sure if this could cause any problems with React's component state life-cycle. FWIW, I haven't been able to produce any obvious issues, but anyone else weighing in would be appreciated.

Reply:
There might a race-condition here if the component has rendered the nodes mapping before it is mutated here, but we can circle back if we start experiencing any issue and should be well aware of what the issue is.